src/bridge/bridge.c

```diff
@@ -173,66 +173,41 @@ send_init_command (CockpitTransport *transport,
   g_bytes_unref (bytes);
 }
 
-static void
-setup_dbus_daemon (gpointer addrfd)
-{
-  g_unsetenv ("G_DEBUG");
-  cockpit_unix_fd_close_all (3, GPOINTER_TO_INT (addrfd));
-}
-
-static GPid
+static GSubprocess *
 start_dbus_daemon (void)
 {
-  GError *error = NULL;
-  GString *address = NULL;
+  g_autoptr(GError) error = NULL;
   gchar *line;
   gsize len;
   gssize ret;
-  GPid pid = 0;
-  gchar *print_address = NULL;
+  GSubprocess *process = NULL;
   int addrfd[2] = { -1, -1 };
-  GSpawnFlags flags;
-
-  gchar *dbus_argv[] = {
-      "dbus-daemon",
-      "--print-address=X",
-      "--session",
-      NULL
-  };
 
   if (pipe (addrfd))
     {
       g_warning ("pipe failed to allocate fds: %m");
-      goto out;
+      return NULL;
     }
 
-  print_address = g_strdup_printf ("--print-address=%d", addrfd[1]);
-  dbus_argv[1] = print_address;
-
   /* The DBus daemon produces useless messages on stderr mixed in */
```

Comment:
also stdout?  I wonder if we couldn't use a stdout pipe: that would be a lot easier than all of this messing about with launchers and pipes and read loops...

Reply:
The current `dbus-daemon` that I have works ok with `--print-address` (which goes to stdout). I'm a bit afraid of dbus-daemon writing anything else there, but given that it is advertised for exactly this use case, let's do this. I do this as a separate commit, just in case.